## Modis folder analysis



In [2]:
import numpy as np
import pandas as pd
import os
import matplotlib.pyplot as plt
from glob import glob
from tqdm import tqdm
from data_loader_analysis import Modis_data_loader
from data_loader_analysis import PM_25_dataloader
from data_loader_analysis import combine_the_data_frames

### Modis data analysis

In [3]:
Modis_data_2018 = Modis_data_loader("MODIS_AOD/merged_data_2018_sorted_dates.csv")

In [4]:
Modis_data_2018.get_head()
# The data is of 365 days latitude and longitude hence the )

latitude  longitude  2018-01-01  2018-01-02  2018-01-03  2018-01-04  \
0     28.414     77.171         NaN       0.322         NaN       0.816   
1     28.414     77.181         NaN       0.305         NaN         NaN   
2     28.414     77.192         NaN       0.293         NaN         NaN   
3     28.414     77.202         NaN       0.305         NaN         NaN   
4     28.414     77.213         NaN       0.319         NaN         NaN   
5     28.414     77.223         NaN       0.331         NaN         NaN   
6     28.424     77.171         NaN       0.328         NaN       0.806   
7     28.424     77.181         NaN       0.328         NaN       0.807   
8     28.424     77.192         NaN       0.317         NaN         NaN   
9     28.424     77.202         NaN       0.307         NaN         NaN   
10    28.424     77.213         NaN       0.341         NaN         NaN   
11    28.424     77.223         NaN       0.348         NaN         NaN   
12    28.424     77.233         NaN       0.327         NaN         NaN   
13    28.424     77.244         NaN       0.310         NaN       0.910   
14    28.434     77.161         NaN       0.393         NaN       0.774   
15    28.434     77.171         NaN       0.356         NaN       0.793   
16    28.434     77.181         NaN       0.327         NaN       0.806   
17    28.434     77.192         NaN       0.311         NaN       0.803   
18    28.434     77.202         NaN       0.300         NaN       0.811   
19    28.434     77.213         NaN       0.314         NaN         NaN   

    2018-01-05  2018-01-06  2018-01-07  2018-01-08  ...  2018-12-22  \
0        1.422       2.179       1.117         NaN  ...       0.552   
1        1.360       2.154       1.069         NaN  ...       0.543   
2        1.337       2.130       1.059         NaN  ...       0.550   
3        1.314       2.129       1.047         NaN  ...       0.570   
4        1.327       2.147       1.050         NaN  ...       0.591   
5        1.362       2.163       1.085         NaN  ...       0.610   
6        1.476       2.159       1.157         NaN  ...       0.546   
7        1.430       2.141       1.116         NaN  ...       0.554   
8        1.372       2.134       1.079         NaN  ...       0.550   
9        1.348       2.115       1.070         NaN  ...       0.549   
10       1.352       2.136       1.092         NaN  ...       0.595   
11       1.376       2.148       1.117         NaN  ...         NaN   
12       1.363       2.138       1.094         NaN  ...       0.612   
13       1.327       2.113       1.078         NaN  ...       0.608   
14       1.614       2.298       1.197         NaN  ...       0.520   
15       1.582       2.208       1.198         NaN  ...       0.533   
16       1.533       2.121       1.157         NaN  ...       0.546   
17       1.434       2.065       1.096         NaN  ...       0.554   
18       1.381       2.060       1.092         NaN  ...       0.552   
19       1.353       2.082       1.115         NaN  ...       0.573   

    2018-12-23  2018-12-24  2018-12-25  2018-12-26  2018-12-27  2018-12-28  \
0        0.933       1.650       0.921         NaN       0.554         NaN   
1        0.936       1.620       0.884         NaN       0.522         NaN   
2        0.954       1.626       0.870         NaN       0.516         NaN   
3        0.973       1.654       0.863         NaN       0.533         NaN   
4        1.009       1.703       0.879         NaN       0.579         NaN   
5        1.056       1.716       0.910         NaN       0.616         NaN   
6        0.980       1.696       1.004         NaN       0.597         NaN   
7        0.961       1.666       0.937         NaN       0.569         NaN   
8        0.951       1.634       0.887         NaN       0.535         NaN   
9        0.965       1.622       0.885         NaN       0.527         NaN   
10       1.036       1.701       0.915         NaN       0.601         NaN   


In [5]:
Modis_data_2018.get_shape()

(1350, 367)

In [6]:
Modis_data_2019 = Modis_data_loader("MODIS_AOD/merged_data_2019_sorted_dates.csv")
Modis_data_2019.get_head()


latitude  longitude  2019-01-01  2019-01-02  2019-01-03  2019-01-04  \
0     28.414     77.171         NaN         NaN         NaN         NaN   
1     28.414     77.181         NaN         NaN         NaN         NaN   
2     28.414     77.192         NaN         NaN         NaN         NaN   
3     28.414     77.202         NaN         NaN         NaN         NaN   
4     28.414     77.213         NaN         NaN         NaN         NaN   
5     28.414     77.223       1.149         NaN         NaN         NaN   
6     28.424     77.171         NaN         NaN         NaN         NaN   
7     28.424     77.181         NaN         NaN         NaN         NaN   
8     28.424     77.192         NaN         NaN         NaN         NaN   
9     28.424     77.202         NaN         NaN         NaN         NaN   
10    28.424     77.213         NaN         NaN         NaN         NaN   
11    28.424     77.223         NaN         NaN         NaN         NaN   
12    28.424     77.233       1.153         NaN         NaN         NaN   
13    28.424     77.244       1.166         NaN         NaN         NaN   
14    28.434     77.161         NaN         NaN         NaN         NaN   
15    28.434     77.171         NaN         NaN         NaN         NaN   
16    28.434     77.181         NaN         NaN         NaN         NaN   
17    28.434     77.192       1.131         NaN         NaN         NaN   
18    28.434     77.202       1.135         NaN         NaN         NaN   
19    28.434     77.213       1.153         NaN         NaN         NaN   

    2019-01-05  2019-01-06  2019-01-07  2019-01-08  ...  2019-12-22  \
0          NaN         NaN         NaN       1.132  ...         NaN   
1        1.930         NaN         NaN       1.105  ...         NaN   
2        1.910         NaN         NaN       1.101  ...         NaN   
3        1.925         NaN       0.709       1.104  ...         NaN   
4        1.959         NaN         NaN       1.135  ...         NaN   
5        1.987         NaN         NaN       1.174  ...         NaN   
6          NaN         NaN         NaN       1.182  ...         NaN   
7        2.004         NaN         NaN       1.136  ...         NaN   
8        1.958         NaN         NaN       1.106  ...         NaN   
9        1.938         NaN         NaN       1.104  ...         NaN   
10       1.992         NaN         NaN       1.159  ...         NaN   
11       2.006         NaN         NaN       1.186  ...         NaN   
12       2.008         NaN         NaN       1.205  ...         NaN   
13       1.995         NaN       0.743       1.185  ...         NaN   
14         NaN         NaN         NaN       1.250  ...         NaN   
15         NaN         NaN         NaN       1.246  ...         NaN   
16       2.063         NaN         NaN       1.222  ...         NaN   
17       1.987         NaN         NaN       1.146  ...         NaN   
18       1.961         NaN         NaN       1.136  ...         NaN   
19       1.967         NaN         NaN       1.143  ...         NaN   

    2019-12-23  2019-12-24  2019-12-25  2019-12-26  2019-12-27  2019-12-28  \
0          NaN         NaN         NaN       0.531         NaN         NaN   
1          NaN         NaN         NaN       0.658         NaN         NaN   
2          NaN         NaN         NaN       0.636         NaN         NaN   
3          NaN         NaN         NaN       0.608         NaN         NaN   
4          NaN         NaN         NaN       0.615         NaN         NaN   
5          NaN         NaN         NaN       0.633         NaN         NaN   
6          NaN         NaN         NaN       0.634         NaN         NaN   
7          NaN         NaN         NaN       0.609         NaN         NaN   
8          NaN         NaN         NaN       0.580         NaN         NaN   
9          NaN         NaN         NaN       0.629         NaN         NaN   
10         NaN         NaN         NaN       0.692         NaN         NaN   


In [7]:
Modis_data_2019.get_shape()

(1350, 367)

In [8]:
Modis_data_2020 = Modis_data_loader("MODIS_AOD/merged_data_2020_sorted_dates.csv")
Modis_data_2020.get_head()

latitude  longitude  2020-01-01  2020-01-02  2020-01-03  2020-01-04  \
0     28.414     77.171         NaN         NaN       1.428       1.092   
1     28.414     77.181         NaN         NaN       1.389       1.077   
2     28.414     77.192         NaN         NaN       1.325       1.085   
3     28.414     77.202         NaN         NaN       1.306       1.106   
4     28.414     77.213         NaN         NaN       1.312       1.157   
5     28.414     77.223         NaN         NaN       1.301         NaN   
6     28.424     77.171         NaN         NaN       1.462       1.137   
7     28.424     77.181         NaN         NaN       1.448       1.114   
8     28.424     77.192         NaN         NaN       1.423       1.101   
9     28.424     77.202         NaN         NaN       1.378       1.099   
10    28.424     77.213         NaN         NaN       1.356       1.181   
11    28.424     77.223         NaN         NaN       1.343       1.239   
12    28.424     77.233         NaN         NaN       1.308       1.275   
13    28.424     77.244         NaN         NaN       1.263       1.245   
14    28.434     77.161         NaN         NaN       1.459       1.191   
15    28.434     77.171         NaN         NaN       1.471       1.186   
16    28.434     77.181         NaN         NaN       1.466       1.172   
17    28.434     77.192         NaN         NaN       1.431       1.112   
18    28.434     77.202         NaN         NaN       1.420       1.096   
19    28.434     77.213         NaN         NaN       1.407       1.103   

    2020-01-05  2020-01-06  2020-01-07  2020-01-08  ...  2020-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12         NaN         NaN         NaN         NaN  ...         NaN   
13         NaN         NaN         NaN         NaN  ...         NaN   
14         NaN         NaN         NaN         NaN  ...         NaN   
15         NaN         NaN         NaN         NaN  ...         NaN   
16         NaN         NaN         NaN         NaN  ...         NaN   
17         NaN         NaN         NaN         NaN  ...         NaN   
18         NaN         NaN         NaN         NaN  ...         NaN   
19         NaN         NaN         NaN         NaN  ...         NaN   

    2020-12-23  2020-12-24  2020-12-25  2020-12-26  2020-12-27  2020-12-28  \
0        0.491         NaN         NaN         NaN       0.390       0.531   
1        0.493         NaN         NaN         NaN       0.368       0.515   
2        0.482         NaN         NaN         NaN       0.363       0.506   
3        0.483         NaN         NaN         NaN       0.375       0.505   
4        0.495         NaN         NaN         NaN       0.392       0.532   
5        0.519         NaN         NaN         NaN       0.406       0.565   
6        0.504         NaN         NaN         NaN       0.426       0.564   
7        0.513         NaN         NaN         NaN       0.407       0.538   
8        0.508         NaN       0.583         NaN       0.372       0.513   
9        0.491         NaN       0.549         NaN       0.358       0.500   
10       0.517         NaN         NaN         NaN       0.398       0.550   


In [9]:
Modis_data_2020.get_shape()

(1350, 368)

In [10]:
Modis_data_2021 = Modis_data_loader("MODIS_AOD/merged_data_2021_sorted_dates.csv")
Modis_data_2021.get_head()

latitude  longitude  2021-01-01  2021-01-02  2021-01-03  2021-01-04  \
0     28.414     77.171       0.586         NaN         NaN         NaN   
1     28.414     77.181       0.570         NaN         NaN         NaN   
2     28.414     77.192       0.561         NaN         NaN         NaN   
3     28.414     77.202       0.581         NaN         NaN         NaN   
4     28.414     77.213       0.595         NaN         NaN         NaN   
5     28.414     77.223         NaN         NaN         NaN         NaN   
6     28.424     77.171       0.623         NaN         NaN         NaN   
7     28.424     77.181       0.608         NaN         NaN         NaN   
8     28.424     77.192       0.583         NaN         NaN         NaN   
9     28.424     77.202       0.570         NaN         NaN         NaN   
10    28.424     77.213         NaN         NaN         NaN         NaN   
11    28.424     77.223         NaN         NaN         NaN         NaN   
12    28.424     77.233       0.597         NaN         NaN         NaN   
13    28.424     77.244       0.583         NaN         NaN         NaN   
14    28.434     77.161       0.601         NaN         NaN         NaN   
15    28.434     77.171       0.622         NaN         NaN         NaN   
16    28.434     77.181       0.648         NaN         NaN         NaN   
17    28.434     77.192       0.615         NaN         NaN         NaN   
18    28.434     77.202       0.594         NaN         NaN         NaN   
19    28.434     77.213       0.610         NaN         NaN         NaN   

    2021-01-05  2021-01-06  2021-01-07  2021-01-08  ...  2021-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12         NaN         NaN         NaN         NaN  ...         NaN   
13         NaN         NaN         NaN         NaN  ...         NaN   
14         NaN         NaN         NaN         NaN  ...         NaN   
15         NaN         NaN         NaN         NaN  ...         NaN   
16         NaN         NaN         NaN         NaN  ...         NaN   
17         NaN         NaN         NaN         NaN  ...         NaN   
18         NaN         NaN         NaN         NaN  ...         NaN   
19         NaN         NaN         NaN         NaN  ...         NaN   

    2021-12-23  2021-12-24  2021-12-25  2021-12-26  2021-12-27  2021-12-28  \
0        1.931       1.516         NaN         NaN         NaN         NaN   
1        1.902       1.527         NaN         NaN         NaN         NaN   
2        1.869       1.577         NaN         NaN         NaN         NaN   
3        1.874       1.651         NaN         NaN         NaN         NaN   
4        1.892       1.589         NaN         NaN         NaN         NaN   
5        1.898       1.672         NaN         NaN         NaN         NaN   
6          NaN       1.573         NaN         NaN         NaN         NaN   
7        1.923       1.415         NaN         NaN         NaN         NaN   
8        1.890       1.510         NaN         NaN         NaN         NaN   
9        1.846       1.607         NaN         NaN         NaN         NaN   
10       1.859       1.669         NaN         NaN         NaN         NaN   


In [11]:
Modis_data_2021 = Modis_data_loader("MODIS_AOD/merged_data_2021_sorted_dates.csv")
Modis_data_2021.get_head()

latitude  longitude  2021-01-01  2021-01-02  2021-01-03  2021-01-04  \
0     28.414     77.171       0.586         NaN         NaN         NaN   
1     28.414     77.181       0.570         NaN         NaN         NaN   
2     28.414     77.192       0.561         NaN         NaN         NaN   
3     28.414     77.202       0.581         NaN         NaN         NaN   
4     28.414     77.213       0.595         NaN         NaN         NaN   
5     28.414     77.223         NaN         NaN         NaN         NaN   
6     28.424     77.171       0.623         NaN         NaN         NaN   
7     28.424     77.181       0.608         NaN         NaN         NaN   
8     28.424     77.192       0.583         NaN         NaN         NaN   
9     28.424     77.202       0.570         NaN         NaN         NaN   
10    28.424     77.213         NaN         NaN         NaN         NaN   
11    28.424     77.223         NaN         NaN         NaN         NaN   
12    28.424     77.233       0.597         NaN         NaN         NaN   
13    28.424     77.244       0.583         NaN         NaN         NaN   
14    28.434     77.161       0.601         NaN         NaN         NaN   
15    28.434     77.171       0.622         NaN         NaN         NaN   
16    28.434     77.181       0.648         NaN         NaN         NaN   
17    28.434     77.192       0.615         NaN         NaN         NaN   
18    28.434     77.202       0.594         NaN         NaN         NaN   
19    28.434     77.213       0.610         NaN         NaN         NaN   

    2021-01-05  2021-01-06  2021-01-07  2021-01-08  ...  2021-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12         NaN         NaN         NaN         NaN  ...         NaN   
13         NaN         NaN         NaN         NaN  ...         NaN   
14         NaN         NaN         NaN         NaN  ...         NaN   
15         NaN         NaN         NaN         NaN  ...         NaN   
16         NaN         NaN         NaN         NaN  ...         NaN   
17         NaN         NaN         NaN         NaN  ...         NaN   
18         NaN         NaN         NaN         NaN  ...         NaN   
19         NaN         NaN         NaN         NaN  ...         NaN   

    2021-12-23  2021-12-24  2021-12-25  2021-12-26  2021-12-27  2021-12-28  \
0        1.931       1.516         NaN         NaN         NaN         NaN   
1        1.902       1.527         NaN         NaN         NaN         NaN   
2        1.869       1.577         NaN         NaN         NaN         NaN   
3        1.874       1.651         NaN         NaN         NaN         NaN   
4        1.892       1.589         NaN         NaN         NaN         NaN   
5        1.898       1.672         NaN         NaN         NaN         NaN   
6          NaN       1.573         NaN         NaN         NaN         NaN   
7        1.923       1.415         NaN         NaN         NaN         NaN   
8        1.890       1.510         NaN         NaN         NaN         NaN   
9        1.846       1.607         NaN         NaN         NaN         NaN   
10       1.859       1.669         NaN         NaN         NaN         NaN   


In [12]:
Modis_data_2021.get_shape()

(1350, 367)

In [13]:
Modis_data_2022 = Modis_data_loader("MODIS_AOD/merged_data_2022_sorted_dates.csv")
Modis_data_2022.get_head()

latitude  longitude  2021-12-31  2022-01-02  2022-01-03  2022-01-04  \
0     28.414     77.171         NaN       1.001         NaN         NaN   
1     28.414     77.181         NaN       0.898         NaN         NaN   
2     28.414     77.192         NaN       0.883         NaN         NaN   
3     28.414     77.202         NaN       0.867         NaN         NaN   
4     28.414     77.213         NaN       0.860         NaN         NaN   
5     28.414     77.223         NaN       0.868         NaN         NaN   
6     28.424     77.171         NaN       1.205         NaN         NaN   
7     28.424     77.181         NaN       1.024         NaN         NaN   
8     28.424     77.192         NaN       0.919         NaN         NaN   
9     28.424     77.202         NaN       0.895         NaN         NaN   
10    28.424     77.213         NaN       0.904         NaN         NaN   
11    28.424     77.223         NaN       0.922         NaN         NaN   
12    28.424     77.233         NaN       0.950         NaN         NaN   
13    28.424     77.244         NaN       0.965         NaN         NaN   
14    28.434     77.161         NaN       1.448         NaN         NaN   
15    28.434     77.171         NaN       1.352         NaN         NaN   
16    28.434     77.181         NaN       1.197         NaN         NaN   
17    28.434     77.192         NaN       0.943         NaN         NaN   
18    28.434     77.202         NaN       0.913         NaN         NaN   
19    28.434     77.213         NaN       0.909         NaN         NaN   

    2022-01-05  2022-01-06  2022-01-07  2022-01-08  ...  2022-12-22  \
0          NaN         NaN         NaN         NaN  ...         NaN   
1          NaN         NaN         NaN         NaN  ...         NaN   
2          NaN         NaN         NaN         NaN  ...         NaN   
3          NaN         NaN         NaN         NaN  ...         NaN   
4          NaN         NaN         NaN         NaN  ...         NaN   
5          NaN         NaN         NaN         NaN  ...         NaN   
6          NaN         NaN         NaN         NaN  ...         NaN   
7          NaN         NaN         NaN         NaN  ...         NaN   
8          NaN         NaN         NaN         NaN  ...         NaN   
9          NaN         NaN         NaN         NaN  ...         NaN   
10         NaN         NaN         NaN         NaN  ...         NaN   
11         NaN         NaN         NaN         NaN  ...         NaN   
12         NaN         NaN         NaN         NaN  ...         NaN   
13         NaN         NaN         NaN         NaN  ...         NaN   
14         NaN         NaN         NaN         NaN  ...         NaN   
15         NaN         NaN         NaN         NaN  ...         NaN   
16         NaN         NaN         NaN         NaN  ...         NaN   
17         NaN         NaN         NaN         NaN  ...         NaN   
18         NaN         NaN         NaN         NaN  ...         NaN   
19         NaN         NaN         NaN         NaN  ...         NaN   

    2022-12-23  2022-12-24  2022-12-25  2022-12-26  2022-12-27  2022-12-28  \
0        0.486         NaN         NaN         NaN       0.452         NaN   
1        0.513         NaN         NaN         NaN       0.477         NaN   
2        0.533         NaN         NaN         NaN       0.490         NaN   
3        0.545         NaN         NaN         NaN       0.505         NaN   
4        0.557         NaN         NaN         NaN       0.512         NaN   
5        0.592         NaN         NaN         NaN         NaN         NaN   
6        0.443         NaN         NaN         NaN       0.425         NaN   
7        0.486         NaN         NaN         NaN       0.461         NaN   
8        0.509         NaN         NaN         NaN       0.476         NaN   
9        0.534         NaN         NaN         NaN       0.489         NaN   
10         NaN         NaN         NaN         NaN       0.508         NaN   


In [14]:
Modis_data_2022.get_shape()

(1350, 367)

In [15]:
Modis_data_2023 = Modis_data_loader("MODIS_AOD/merged_data_2023_sorted_dates.csv")
Modis_data_2023.get_head()

latitude  longitude  2023-01-01  2023-01-02  2023-01-03  2023-01-04  \
0     28.414     77.171       0.317         NaN       0.567       0.339   
1     28.414     77.181       0.329         NaN       0.573       0.349   
2     28.414     77.192       0.334         NaN       0.571       0.363   
3     28.414     77.202       0.339         NaN       0.588       0.368   
4     28.414     77.213       0.345         NaN       0.605       0.373   
5     28.414     77.223       0.346         NaN       0.604       0.381   
6     28.424     77.171       0.316         NaN       0.571       0.332   
7     28.424     77.181       0.329         NaN       0.578       0.342   
8     28.424     77.192       0.326         NaN       0.579       0.355   
9     28.424     77.202       0.329         NaN       0.577       0.371   
10    28.424     77.213       0.341         NaN       0.619       0.373   
11    28.424     77.223         NaN         NaN       0.619       0.370   
12    28.424     77.233       0.323         NaN       0.604       0.370   
13    28.424     77.244       0.322         NaN       0.595       0.358   
14    28.434     77.161       0.275         NaN       0.541       0.352   
15    28.434     77.171       0.298         NaN       0.558       0.352   
16    28.434     77.181       0.315         NaN       0.575       0.356   
17    28.434     77.192       0.323         NaN       0.582       0.363   
18    28.434     77.202       0.327         NaN       0.583       0.381   
19    28.434     77.213       0.333         NaN       0.601       0.379   

    2023-01-05  2023-01-06  2023-01-07  2023-01-08  ...  2023-09-16  \
0        0.375       0.322         NaN         NaN  ...       1.128   
1        0.375       0.336         NaN         NaN  ...       1.141   
2        0.373       0.342         NaN         NaN  ...       1.142   
3        0.385       0.338         NaN         NaN  ...       1.175   
4        0.398       0.355         NaN         NaN  ...       1.211   
5        0.397       0.364         NaN         NaN  ...         NaN   
6        0.377       0.330         NaN         NaN  ...         NaN   
7        0.386       0.345         NaN         NaN  ...       1.153   
8        0.379       0.339         NaN         NaN  ...       1.146   
9        0.373       0.340         NaN         NaN  ...       1.132   
10       0.403       0.358         NaN         NaN  ...       1.209   
11       0.403       0.364         NaN         NaN  ...       1.247   
12       0.384       0.364         NaN         NaN  ...         NaN   
13       0.368       0.320         NaN         NaN  ...         NaN   
14       0.331       0.262         NaN         NaN  ...         NaN   
15       0.353       0.304         NaN         NaN  ...         NaN   
16       0.375       0.345         NaN         NaN  ...       1.169   
17       0.372       0.344         NaN         NaN  ...       1.120   
18       0.367       0.348         NaN         NaN  ...       1.113   
19       0.378       0.342         NaN         NaN  ...       1.146   

    2023-09-17  2023-09-18  2023-09-19  2023-09-20  2023-09-21  2023-09-22  \
0        0.707         NaN         NaN         NaN       0.438       0.465   
1        0.648         NaN         NaN         NaN       0.412       0.391   
2        0.638         NaN         NaN         NaN       0.384       0.394   
3        0.644         NaN         NaN         NaN       0.404       0.417   
4        0.702         NaN         NaN         NaN       0.431       0.472   
5        0.725         NaN         NaN         NaN       0.428       0.498   
6        0.793         NaN         NaN         NaN       0.451       0.533   
7        0.753         NaN         NaN         NaN       0.446       0.475   
8        0.693         NaN         NaN         NaN       0.410       0.411   
9        0.687         NaN         NaN         NaN       0.362       0.394   
10       0.761         NaN         NaN         NaN       0.422       0.474   


In [16]:
Modis_data_2023.get_shape()

(1350, 250)

### Latitude & Longitude data sets




In [17]:
Lat_Long_data = Modis_data_loader("Stations_Lat_Long/stn_extrafeat.csv")
stations = Lat_Long_data.data[["Latitude", "Longitude"]]
stations

Latitude  Longitude
0   28.815329  77.153010
1   28.646835  77.316032
2   28.695381  77.181665
3   28.470691  77.109936
4   28.776200  77.051074
5   28.725650  77.201157
6   28.551201  77.273574
7   28.656756  77.227234
8   28.750050  77.111261
9   28.498571  77.264840
10  28.571027  77.071901
11  28.655602  77.285932
12  28.562776  77.118005
13  28.681174  77.302523
14  28.631694  77.249439
15  28.732820  77.170633
16  28.580280  77.233829
17  28.588333  77.221667
18  28.591825  77.227307
19  28.611281  77.237738
20  28.636429  77.201067
21  28.684678  77.076574
22  28.609090  77.032541
23  28.570173  76.933762
24  28.822836  77.101981
25  28.567890  77.250515
26  28.657381  77.158545
27  28.530785  77.271255
28  28.623748  77.287205
29  28.563262  77.186937
30  28.645000  77.167000
31  28.639645  77.146262
32  28.674045  77.131023
33  28.732528  77.119920
34  28.651478  77.147311
35  28.550425  77.215938
36  28.710508  77.249485
37  28.531346  77.190156
38  28.672342  77.315260
39  28.699793  77.165453

## PM_2.5 data set


In [18]:
# Shape of the raw dataset.
PM_data = PM_25_dataloader()
PM_data.get_shape()


(16, 40, 8760)

##### Shape of the data set is `(8760, 40, 16)` from here we have seen that there are 8760 hours in a year and 40 stations and 16 features in the data set but we have extracted the PM_25 data values for 40 stations and 8760 time stamps.

In [20]:
PM_data = PM_25_dataloader()
PM_data = PM_data.get_data()
PM_data_with_Lat_long = combine_the_data_frames(stations, PM_data)
PM_data_with_Lat_long.head()

Latitude  Longitude           0           1           2           3  \
0  28.815329  77.153010  265.333344  341.500000  323.333344  340.000000   
1  28.646835  77.316032  571.666687  366.833344  421.250000  335.500000   
2  28.695381  77.181665  437.833344  327.666656  427.000000  376.500000   
3  28.470691  77.109936  550.376648  619.489990  465.200012  304.130005   
4  28.776200  77.051074  688.000000  417.333344  382.333344  451.333344   

            4           5           6           7  ...        8750  \
0  132.166672  106.833336  162.166672  241.666672  ...  272.333344   
1  154.833328  106.833336  256.416656  725.666687  ...  268.500000   
2  169.666672  157.833328  437.000000  749.166687  ...  338.500000   
3  143.858337  122.298332  203.143326  323.450012  ...  275.351654   
4  232.000000  163.000000  388.000000  746.755066  ...  351.166656   

         8751        8752        8753        8754        8755        8756  \
0  369.333344  335.000000  275.666656  261.333344  325.666656  226.000000   
1  268.166656  262.666656  225.166672  243.000000  240.666672  171.833328   
2  382.833344  349.666656  273.833344  259.166656  245.833328  196.333328   
3  283.111664  237.164993  208.151672  192.948334  185.583328  161.544998   
4  447.666656  390.333344  300.500000  270.666656  257.166656  228.500000   

         8757        8758        8759  
0  223.666672  279.000000  343.000000  
1  169.500000  302.833344  456.500000  
2  163.333328  325.333344  476.000000  
3  161.471664  261.464996  244.146667  
4  218.500000  391.500000  375.833344  

[5 rows x 8762 columns]